internal/controller/finbackupconfig_controller_test.go

```diff
@@ -0,0 +1,299 @@
+			defer func() { DeletePVCAndPV(ctx, pvc.Namespace, pvc.Name) }()
+
+			pod := createControllerPod("pod-mismatch", namespace)
+			Expect(k8sClient.Create(ctx, pod)).NotTo(HaveOccurred())
+			defer func() { _ = k8sClient.Delete(ctx, pod) }()
+			Expect(os.Setenv("POD_NAME", pod.Name)).To(Succeed())
+			Expect(os.Setenv("POD_NAMESPACE", pod.Namespace)).To(Succeed())
+			Expect(os.Setenv("CREATE_FINBACKUP_JOB_SERVICE_ACCOUNT", "test-sa")).To(Succeed())
+
+			fbc := &finv1.FinBackupConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: "fbc-mismatch", Namespace: namespace},
+				Spec:       finv1.FinBackupConfigSpec{PVC: pvc.Name, Schedule: "0 2 * * *", Suspend: false},
+			}
+			Expect(k8sClient.Create(ctx, fbc)).NotTo(HaveOccurred())
+			defer func() { _ = k8sClient.Delete(ctx, fbc) }()
+
+			req := ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      fbc.Name,
+					Namespace: fbc.Namespace,
+				},
+			}
+			_, err := reconciler.Reconcile(ctx, req)
+			Expect(err).NotTo(HaveOccurred())
+
+			cronJob := &batchv1.CronJob{}
+			cronJobName := "fbc-" + string(fbc.UID)
+			Consistently(func() error {
+				return k8sClient.Get(ctx, types.NamespacedName{Name: cronJobName, Namespace: namespace}, cronJob)
+			}, 3*time.Second).ShouldNot(Succeed())
+		})
+
+		It("should use overwriteFBCSchedule when reconciler is initialized with it", func() {
+			// Description:
+			//   Verify that the reconciler's overwriteFBCSchedule takes precedence for CronJob schedule.
+			//
+			// Arrange:
+			//   - Create a Reconciler with overwriteFBCSchedule set.
+			//   - Prepare an RBD StorageClass, PVC/PV, controller Pod and env vars.
+			//   - Create a FinBackupConfig.
+			//
+			// Act:
+			//   - Call Reconcile() on the local reconciler.
+			//
+			// Assert:
+			//   - CronJob schedule is the overwriteFBCSchedule value.
+			localReconciler := NewFinBackupConfigReconciler(k8sClient, scheme.Scheme, "15 3 * * *")
+			localReconciler.finCephClusterID = namespace
+
+			sc := NewRBDStorageClass("test-overwrite", namespace, rbdPoolName)
+			Expect(k8sClient.Create(ctx, sc)).NotTo(HaveOccurred())
+			defer func() { _ = k8sClient.Delete(ctx, sc) }()
+
+			pvc, pv := NewPVCAndPV(sc, namespace, "pvc-overwrite", "pv-overwrite", rbdImageName)
+			Expect(k8sClient.Create(ctx, pvc)).NotTo(HaveOccurred())
+			Expect(k8sClient.Create(ctx, pv)).NotTo(HaveOccurred())
+			defer func() { DeletePVCAndPV(ctx, pvc.Namespace, pvc.Name) }()
+
+			pod := createControllerPod("pod-overwrite", namespace)
+			Expect(k8sClient.Create(ctx, pod)).NotTo(HaveOccurred())
+			defer func() { _ = k8sClient.Delete(ctx, pod) }()
+			Expect(os.Setenv("POD_NAME", pod.Name)).To(Succeed())
+			Expect(os.Setenv("POD_NAMESPACE", pod.Namespace)).To(Succeed())
+			Expect(os.Setenv("CREATE_FINBACKUP_JOB_SERVICE_ACCOUNT", "test-sa")).To(Succeed())
+
+			fbc := &finv1.FinBackupConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: "fbc-overwrite", Namespace: namespace},
```

Comment:
Let's use a different namespace from that of the controller Pod to make things more realistic.